terraform/constellation-cluster/variables.tf

```diff
@@ -0,0 +1,89 @@
+variable "csp" {
+  type        = string
+  description = "The CSP to create the cluster in"
+  validation {
+    condition     = var.csp == "aws" || var.csp == "azure" || var.csp == "gcp"
+    error_message = "The CSP must be 'aws', 'azure' or 'gcp'."
+  }
+}
+
+variable "node_groups" {
+  type = map(object({
+    role          = string
+    initial_count = optional(number)
+    instance_type = string
+    disk_size     = number
+    disk_type     = string
+    zone          = string
+  }))
+  description = "A map of node group names to node group configurations."
+  validation {
+    condition     = can([for group in var.node_groups : group.role == "control-plane" || group.role == "worker"])
+    error_message = "The role has to be 'control-plane' or 'worker'."
+  }
+}
+
+variable "name" {
+  type        = string
+  description = "Name used in the cluster's named resources / cluster name"
+}
+
+# STATE related
```

Comment:
```suggestion
```
We should remove these comments here or reword them, as this file will be customer-facing